Client/src/Components/Graph/Graph.tsx

```diff
@@ -31,19 +35,74 @@ export default function Graph(props: IProps) {
   const outerHeight = props && props.outerHeight || 500
   const width = outerWidth - margin.left - margin.right
   const height = outerHeight - margin.top - margin.bottom
+  const extremeBoundaries = props.extremeBoundaries
 
   const [isXLog, setXToggle] = useState(false)
   const [isYLog, setYToggle] = useState(false)
+  const [showSettings, setSettingsToggle] = useState(false)
+  const [xUpperBound, setXUpperBound] = useState(extremeBoundaries.maxX)
+  const [xLowerBound, setXLowerBound] = useState(extremeBoundaries.minX)
+  const [yUpperBound, setYUpperBound] = useState(extremeBoundaries.maxY)
+  const [yLowerBound, setYLowerBound] = useState(extremeBoundaries.minY)
+  // This loops through the selected arrays to find the min/max for both x and y.
 
   const handleXScaleClick = () => {
+    if (xLowerBound <= 0 && !isXLog) {
+      setXLowerBound(1)
```

Comment:
you can use the whole object as a state, that way you only have one state representing the boundaries, you then set it with: setBoundaries({...boundaries, minY: 1 }) etc.

Reply:
I've modified it to now use an object as a state.